### PR TITLE
[pt2] add meta definitions for `resize_` and `resize_as_`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #98282

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -124,6 +126,33 @@ def meta_copy_(self, src, non_blocking=False):
     return self
 
 
+@register_meta(aten.resize_.default)
+def meta_resize_(self, size, *, memory_format=None):
+    if same_shape(self.shape, size):
+        return self  # ignores memory_format
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:132]: This implements behavior from the docs: "Note that memory format of self is going to be unaffected if self.size() matches sizes."

However, ATen always uses `memory_format`. `_resize_impl_` early exits, but `memory_format` is used later anyway:

```cpp
template <typename T>
const Tensor& _resize_(
    const Tensor& self,
    ArrayRef<T> size,
    c10::optional<MemoryFormat> optional_memory_format) {
  auto* self_ = self.unsafeGetTensorImpl();
  // NOLINTNEXTLINE(bugprone-argument-comment)
  _resize_impl_<T>(self_, size, /*strides=*/c10::nullopt, true);
  if (optional_memory_format.has_value()) {
    auto memory_format =
        optional_memory_format.value();
    TORCH_CHECK(
        memory_format != MemoryFormat::Preserve,
        "Unsupported memory format",
        memory_format);
    self_->empty_tensor_restride(memory_format);
  }
  return self;
}
```

```
[ins] In [6]: torch._meta_registrations.meta_resize_(torch.randn(3, 4, 5, 6), (3, 4, 5, 6), memory_format=torch.channels_last).is_contiguous()
Out[6]: True

[ins] In [7]: torch.Tensor.resize_(torch.randn(3, 4, 5, 6), (3, 4, 5, 6), memory_format=torch.channels_last).is_contiguous()
Out[7]: False

[ins] In [8]: torch.Tensor.resize_(torch.randn(3, 4, 5, 6), (3, 4, 5, 6), memory_format=torch.channels_last).is_contiguous(memory_format=torch.channels_last)
Out[8]: True

```

**Comment by nkaretnikov** [comment]
Are these metas needed? There are xfails in `test_proxy_tensor.py` and `resize_` has an ATen meta (which I remove here).

File: aten/src/ATen/native/Resize.cpp [hunk 2]
```diff
@@ -223,14 +215,6 @@ TensorImpl* resize_impl_cpu_(
...
 }
 
-TensorImpl* resize_impl_meta_(
-    TensorImpl* self,
-    c10::SymIntArrayRef size,
-    at::OptionalSymIntArrayRef stride,
-    bool resize_storage = true) {
-  return _resize_impl_(self, size, stride, resize_storage);
-}
-
```
Comment (nkaretnikov) [inline aten/src/ATen/native/Resize.cpp:233]: Was this unused?

File: aten/src/ATen/native/Resize.cpp [hunk 3]
```diff
@@ -261,13 +245,5 @@ const Tensor& resize_(
   return _resize_(self, size, optional_memory_format);
 }
 
-const Tensor& resize__symint(
-    const Tensor& self,
-    c10::SymIntArrayRef size,
-    c10::optional<MemoryFormat> optional_memory_format) {
-  TORCH_INTERNAL_ASSERT(!self.has_names())
-  return _resize_(self, size, optional_memory_format);
-}
```
Comment (nkaretnikov) [inline aten/src/ATen/native/Resize.cpp:270]: Is it cool to remove this and the Meta dispatch from ATen?

**Comment by ezyang** [comment]
I'm not reviewing this

**Pull request closed without merging**